src/memory.c

```diff
@@ -348,19 +354,32 @@ static int memory_read_internal (value_list_t *vl)
 	 * They sum up to the value of Slab, which is available on older & newer
 	 * kernels. So SReclaimable/SUnreclaim are submitted if available, and Slab
 	 * if not. */
-	if (detailed_slab_info)
-		MEMORY_SUBMIT ("used",        mem_used,
-		               "buffered",    mem_buffered,
-		               "cached",      mem_cached,
-		               "free",        mem_free,
-		               "slab_unrecl", mem_slab_unreclaimable,
-		               "slab_recl",   mem_slab_reclaimable);
-	else
+	if (detailed_slab_info) {
+		/* The only condition under which there will be a MemAvailable line is
+		 * on a kernel which also supports detailed slab information. */
+		if (mem_available_info) {
+			MEMORY_SUBMIT ("used",          mem_used,
+			               "buffered",      mem_buffered,
+			               "cached",        mem_cached,
+			               "free",          mem_free,
+			               "available",     mem_available,
```

Comment:
Is this accounted as separate from the rest or is it a part of the overall memory?  From your description, I think the former?  In which case, I think this needs to be done in a separate submit to avoid throwing things off if ValuesPercentage is set to true.